File: _citybox_src/apps/marketplace/ios/Sources/Components/SectionHeader.swift
import SwiftUI

struct SectionHeader: View {
    let title: String
    var actionTitle: String? = "Ver tudo"
    var action: (() -> Void)? = nil

    var body: some View {
        HStack {
            Text(title)
                .font(CBFont.section())
                .foregroundColor(.cbBlack)

            Spacer()

            if let actionTitle = actionTitle, let action = action {
                Button(action: action) {
                    Text(actionTitle)
                        .font(CBFont.body2())
                        .foregroundColor(.cbBlack)
                }
            }
        }
    }
}

File: _citybox_src/apps/marketplace/android/app/src/main/java/com/citybox/ui/components/SectionHeader.kt
package com.citybox.ui.components

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.PlatformTextStyle
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.style.LineHeightStyle
import com.citybox.ui.theme.Black
import com.citybox.ui.theme.CBFont

private val headerTextStyle = TextStyle(
    platformStyle = PlatformTextStyle(includeFontPadding = false),
    lineHeightStyle = LineHeightStyle(
        alignment = LineHeightStyle.Alignment.Center,
        trim = LineHeightStyle.Trim.Both
    )
)

@Composable
fun SectionHeader(
    title: String,
    actionText: String = "Ver tudo",
    onAction: (() -> Unit)? = null,
    modifier: Modifier = Modifier
) {
    Row(
        modifier = modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically
    ) {
        Text(
            text = title,
            style = CBFont.Section.merge(headerTextStyle),
            color = Black
        )
        if (onAction != null) {
            Text(
                text = actionText,
                style = CBFont.Body2.merge(headerTextStyle),
                color = Black,
                modifier = Modifier.clickable(onClick = onAction)
            )
        }
    }
}
